docs/reference/style-guides/node-style-guide.md

```diff
@@ -2,8 +2,186 @@
 title: Node Style Guide
 ---
 
-Current as of `July 18, 2022`
+Current as of `January 18, 2023`
 
-## Introduction
+## Code Standards
 
-Building...
+These code standards apply to new packages in the FxA monorepo, such as:
+
+- `packages/fxa-graphql-api`
+- `packages/fxa-event-broker`
+
+Most of the standards in the code is enforced with [Prettier][2] and [ESLint][1].
+
+### Style
+
+We use the [Google TypeScript Style Guide](https://google.github.io/styleguide/tsguide.html#source-organization) for our base code style.
+
+Code should also follow the [Do's and Don'ts of TypeScript](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html).
+
+### Structure
+
+#### Organize by Feature
+
+(This is present in the Google TypeScript Style Guide, but bears repeating)
+
+Code should be organized by feature, not by type.
+
+Don't:
+
+```
+app/
+├── controllers/
+│   └── auth.ts
+├── models/
+│   └── auth.ts
+├── validators/
+│   └── auth.ts
+```
+
+Do:
+
+```
+app/
+├── auth/
+│   ├── auth.controller.ts
+│   ├── auth.model.ts
+│   └── auth.validator.ts
+```
+
+If a feature has many DTOs, models, etc., a hybrid approach may be used, organized by feature first:
+
+```
+app/
+├── auth/
+│   ├── dto/
+│   ├── model/
+│   |   └── auth.model.ts
+│   ├── auth.controller.ts
+│   └── auth.validator.ts
+```
+
+Tests should be organized alongside the code being tested, with the `.spec.ts` suffix:
+
+```
+app/
+├── auth/
+│   ├── auth.controller.ts
+│   ├── auth.controller.spec.ts
+```
+
+#### NestJS File Naming Conventions
+
+##### File Naming for Class
+
+```typescript
+export class PascalCaseSuffix {} //= pascal-case.suffix.ts
+// Except for suffix, PascalCase to hyphen-case
+class FooBarNaming {} //= foo-bar.naming.ts
+class FooController {} //= foo.controller.ts
+class BarQueryDto {} //= bar-query.dto.ts
+```
+
+#### Interface Naming
+
+```typescript
+// https://stackoverflow.com/questions/541912
+// https://stackoverflow.com/questions/2814805
+interface User {}
+interface CustomUser extends User {}
+interface ThirdCustomUser extends CustomUser {}
+```
+
+#### Index Exporting
+
+```diff
```

Comment:
Further up you use Do and Don't sections and here you're using a diff. Can this be consistent in terms of providing good and bad examples?